packages/api/core/src/util/forge-config.ts

```diff
@@ -104,6 +104,7 @@ export function renderConfigTemplate(dir: string, templateObj: any, obj: any): v
 }
 
 type MaybeESM<T> = T | { default: T };
+type AsyncFn = () => Promise<ForgeConfig>;
```

Comment:
`AsyncForgeConfigGenerator` or something? AsyncFn isn't a super descriptive type name

Reply:
Yeah, that was very generic - modified to `AsyncForgeConfigGenerator`